translator.py

```diff
@@ -0,0 +1,110 @@
+#!/usr/bin/env python3
+"""Translates any text document into an .fb file
+
+First CLI argument should be a text file to be converted.
+Print to `.fb` file extension with ` > output_file.fb`
+
+Example:
+
+./translator.py original_text_file.txt > new_fb_file.fb
+
+Check translation with ./code_runner.py new_fb_file.fb"""
+
+import sys
+
+
+class Translator():
+    """Prepares common FizzBuzzLang functions and retrieves file text content
+
+    ---
+    Attributes:
+        look_up: dict -- self-populating dictionary of finary refs
+        INPUT: str -- text content of file passed in CLI
+        STORE_CHAR: str -- allows storing of character at data-space location
+        MOVE_FORWARD: str -- move pointer forward one space in data-space
+        STORE_POS_FIZZ: str -- stores current data-space location in FIZZ loc.
+        END_OF_CODE: str -- prints data
+        TERMINATE_COMMAND: str -- termination command to end fbi
+        FBL_TEXT_ARRAY: list -- INPUT written in finary
+    """
+
+    def __init__(self) -> None:
+        self.look_up = {}
+        self.INPUT = self._retrieve_file()
+        self.STORE_CHAR = "BUZZ FIZZBUZZ FIZZBUZZ "
+        self.MOVE_FORWARD = "FIZZ FIZZ FIZZ"
+        self.STORE_POS_FIZZ = "FIZZ FIZZBUZZ FIZZ"
+        self.END_OF_CODE = """BUZZ FIZZBUZZ FIZZBUZZ BUZZ FIZZ BUZZ FIZZ
+FIZZ FIZZBUZZ FIZZBUZZ FIZZ
+FIZZBUZZ FIZZ FIZZBUZZBUZZ
+BUZZ BUZZ
+FIZZ FIZZ FIZZ
+FIZZBUZZ BUZZ FIZZ FIZZBUZZBUZZ"""
+        self.TERMINATE_COMMAND = "FIZZBUZZ FIZZBUZZ"
+        self.FBL_TEXT_ARRAY = self._convert_to_fbl()
+
+    def _retrieve_file(self):
+        """Retrieves text from file
+
+        Accepts CLI argument
+
+        ---
+        returns:
+            str: -- document contents as string
+
+        raises:
+            IndexError: -- if no file name provided
+            FileNotFoundError: -- if file is not found
+
+            All errors will terminate program
+        """
+
+        try:
+            file_name = sys.argv[1]
+        except IndexError:
+            print("Error, no file argument given when running file.")
+            print("Program will terminate.")
+            exit()
```

Comment:
I'd also be very much against an internal function calling `exit`. I would recommend you raise a custom exception and let the caller figure it out

Reply:
(Or, preferably, just not deal with command line arguments here)

Reply:
I would like to handle any errors elegantly, perhaps I should change from an immediate exit to an `input`, this way a user would have an opportunity to add the missing argument without re-initiating the program. I could add some logic to allow the user to choose to exit the program (if for some reason they decide against the operation?), perhaps this would be unnecessary!